File: 백준/Silver/19941. 햄버거 분배/햄버거 분배.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const [n, k] = input[0].split(" ").map(Number)
const data = input[1].split("")

const visited = data.map((val) => (val === "P" ? true : false))
let answer = 0
for (let i = 0; i < n; i++) {
  const cur = data[i]
  if (cur === "H") continue
  for (let j = i - k; j <= i+k; j++) {
    if (j >= 0 && j<n && !visited[j]) {
      visited[j] = true
      answer += 1
      break
    }
  }
}
console.log(answer)

File: 백준/Silver/19941. 햄버거 분배/햄버거 분배.py
import sys
input = sys.stdin.readline

n,k = map(int,input().split(' '))
data = list(input())
answer =0

for i in range(0,n):
  if data[i] == 'P':
    for j in range(i-k,i+k+1):
      if 0<=j <n and data[j] == 'H':
        answer += 1
        data[j] = 'X'
        break
print(answer)